src/zenml/client.py

```diff
@@ -4139,20 +4140,35 @@ def get_artifact_version(
         Returns:
             The artifact version.
         """
+        from zenml import get_step_context
+
         if cll := client_lazy_loader(
             method_name="get_artifact_version",
             name_id_or_prefix=name_id_or_prefix,
             version=version,
             hydrate=hydrate,
         ):
             return cll  # type: ignore[return-value]
-        return self._get_entity_version_by_id_or_name_or_prefix(
+
+        artifact = self._get_entity_version_by_id_or_name_or_prefix(
             get_method=self.zen_store.get_artifact_version,
             list_method=self.list_artifact_versions,
             name_id_or_prefix=name_id_or_prefix,
             version=version,
             hydrate=hydrate,
         )
+        try:
```

Comment:
I'm wondering whether we should actually track this here? Is this really "loading" an input artifact if I just get the response to check some metadata? I guess I'm still using it in some way, even if I don't access the data?

In any case, having this here means we should remove the identical code in `load_artifact`, otherwise we just do the same exact call twice

Reply:
I would say - yes, we should track this, even if one uses metadata only.
I removed the duplication from `load_artifact` funciton.